skyportal/handlers/api/source.py

```diff
@@ -288,3 +291,101 @@ def get(self, source_id):
+            minimum: 0
+            maximum: 10
+          description: |
+            Requested number of offset stars (set to zero to get starlist
+            of just the source itself)
+        - in: query
+          name: obstime_isoformat
+          nullable: True
+          schema:
+            type: string
+          description: |
+            datetime of observation in isoformat (e.g. 2020-12-30T12:34:10)
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+        source = Source.get_if_owned_by(source_id, self.current_user)
+        if source is None:
+            return self.error('Invalid source ID.')
+        if not set(source.groups).intersection(set(self.current_user.groups)):
+            return self.error('Inadequate permissions.')
+
+        facility = self.get_query_argument('facility', 'Keck')
+        how_many = self.get_query_argument('how_many', '3')
+        obstime_isoformat = \
+            self.get_query_argument('obstime_isoformat',
+                                    datetime.datetime.utcnow().isoformat())
+
+        if facility not in facility_parameters:
+            return self.error('Invalid facility')
+
+        radius_degrees = facility_parameters[facility]["radius_degrees"]
+        mag_limit = facility_parameters[facility]["mag_limit"]
+        min_sep_arcsec = facility_parameters[facility]["min_sep_arcsec"]
+        mag_min = facility_parameters[facility]["mag_min"]
+
+        try:
+            how_many = int(how_many)
+        except ValueError:
+            # could not handle inputs
+            return self.error('Error converting some input parameters')
+
+        try:
+            r, query_string, remaining_searches, noffsets = \
+                get_nearby_offset_stars(source.ra, source.dec,
+                                        source_id,
+                                        how_many=how_many,
+                                        radius_degrees=radius_degrees,
+                                        mag_limit=mag_limit,
+                                        min_sep_arcsec=min_sep_arcsec,
+                                        starlist_type=facility,
+                                        mag_min=mag_min,
+                                        obstime_isoformat=obstime_isoformat,
+                                        remaining_searches=1)
+
+        except ValueError:
+            return self.error('Error calling get_nearby_offset_stars')
+
+        return self.success(data={'facility': facility,
+                                  'starlist': r.replace(" ", "&nbsp;"),
+                                  'ra': source.ra,
+                                  'dec': source.dec, 'noffsets': noffsets,
+                                  'remaining_searches': remaining_searches,
```

Comment:
Is this information useful to the user? Or maybe the question is rather: what does the user want to know here?  Perhaps number of queries issued?

Reply:
Correct. See below.